src/ModuleApi.ts

```diff
@@ -46,14 +47,14 @@ export interface ModuleApi {
 
     /**
      * Opens a dialog in the client.
-     * @param title The title of the dialog
+     * @param titleOrOptions The title of the dialog, or an object that configures the dialog.
      * @param body The function which creates a body component for the dialog.
      * @param props Optional props to provide to the dialog.
      * @returns Whether the user submitted the dialog or closed it, and the model returned by the
      * dialog component if submitted.
      */
     openDialog<M extends object, P extends DialogProps = DialogProps, C extends DialogContent<P> = DialogContent<P>>(
-        title: string,
+        titleOrOptions: string | ModuleUiDialogOptions,
```

Comment:
Since these can now be changed by the dialog, let's reflect that in the name.

```suggestion
        initialTitleOrOptions: string | ModuleUiDialogOptions,
```